### ISSUE TITLE
bunyan 'src' handling is broken in node v7

### ISSUE THREAD

**Issue by trentm** [trentm/node-bunyan#454]
The 'src' test file fails:

```
$ ./node_modules/.bin/nodeunit test/src.test.js

src.test.js
✖ src

AssertionError
    at Object.equal (/Users/trentm/tm/node-bunyan/node_modules/nodeunit/lib/types.js:83:39)
    at /Users/trentm/tm/node-bunyan/test/src.test.js:52:11
    at Array.forEach (native)
    at Object.<anonymous> (/Users/trentm/tm/node-bunyan/test/src.test.js:49:10)
    at Object._ (/Users/trentm/tm/node-bunyan/test/tap4nodeunit.js:56:20)
    at Object.<anonymous> (/Users/trentm/tm/node-bunyan/node_modules/nodeunit/lib/core.js:236:16)
    at /Users/trentm/tm/node-bunyan/node_modules/nodeunit/lib/core.js:236:16
    at Object.exports.runTest (/Users/trentm/tm/node-bunyan/node_modules/nodeunit/lib/core.js:70:9)
    at /Users/trentm/tm/node-bunyan/node_modules/nodeunit/lib/core.js:118:25
    at /Users/trentm/tm/node-bunyan/node_modules/nodeunit/deps/async.js:513:13
    at iterate (/Users/trentm/tm/node-bunyan/node_modules/nodeunit/deps/async.js:123:13)
    at async.forEachSeries (/Users/trentm/tm/node-bunyan/node_modules/nodeunit/deps/async.js:139:9)
    at _concat (/Users/trentm/tm/node-bunyan/node_modules/nodeunit/deps/async.js:512:9)
    at Object.concatSeries (/Users/trentm/tm/node-bunyan/node_modules/nodeunit/deps/async.js:152:23)
    at Object.exports.runSuite (/Users/trentm/tm/node-bunyan/node_modules/nodeunit/lib/core.js:96:11)
    at Object.exports.runModule (/Users/trentm/tm/node-bunyan/node_modules/nodeunit/lib/core.js:158:13)

...

FAILURES: 9/14 assertions failed (43ms)
```

**Comment by trentm** [comment]
From https://nodejs.org/en/download/releases/ here is the v8 version jump
in node v7:

```
                                        V8 VERSION
Node.js v7.0.0 		2016-10-25 	5.4.500.36
Node.js v6.9.1 	Boron 	2016-10-19 	5.1.281.84
```

V8 release notes changes, FWIW:
- http://v8project.blogspot.ca/2016/09/v8-release-54.html
- http://v8project.blogspot.ca/2016/07/v8-release-53.html
- http://v8project.blogspot.ca/2016/06/release-52.html

v8 api changes: https://docs.google.com/document/d/1g8JFi8T_oAE_7uAri7Njtig7fKaPDfotU6huOa1alds/edit


With a quick look it feels like it is this bug:
https://bugs.chromium.org/p/v8/issues/detail?id=5293

While that is closed "WontFix", it is only one of four forms that fails to
execute an overwritten `Error.prepareStackTrace`. Bunyan's impl can change to
one of the other forms.

**Comment by trentm** [comment]
published in bunyan@1.8.4
